src/test/java/org/broadinstitute/hellbender/engine/spark/JoinReadsWithRefBasesUnitTest.java

```diff
@@ -0,0 +1,72 @@
+package org.broadinstitute.hellbender.engine.spark;
+
+import com.google.api.services.genomics.model.Read;
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import com.google.cloud.dataflow.sdk.values.KV;
+import htsjdk.samtools.SAMRecord;
+import junit.framework.TestCase;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.dataflow.ReadsPreprocessingPipelineTestData;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.RefWindowFunctions;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.ReferenceDataflowSource;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import org.broadinstitute.hellbender.utils.test.FakeReferenceSource;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.*;
+
+public class JoinReadsWithRefBasesUnitTest extends TestCase {
```

Comment:
Add "spark" to the names of these classes, so they don't get confused with the similarly named dataflow equivalents (here, `PairReadsWithRefBasesUnitTest`) -- remember that we often invoke test classes by name using `-Dtest.single=X`